SOPS 3.0.0 does not support encryption contexts with KMS keys

Create a new file with SOPS using AWS KMS with an encryption context:

```bash
sops --kms arn:blah --encryption-context something:value,key:value2 secrets.yaml
```

Expect: Editor to open for new file
Actual:

```
panic: assignment to entry in nil map

goroutine 1 [running]:
go.mozilla.org/sops/keyservice.KeyFromMasterKey(0x1abede0, 0xc420083360, 0x0, 0x0)
	/private/tmp/sops-20171011-79678-1wrcwgy/sops-3.0.0/src/go.mozilla.org/sops/keyservice/keyservice.go:38 +0x163
go.mozilla.org/sops.(*Metadata).UpdateMasterKeysWithKeyServices(0xc4201c0518, 0xc420015140, 0x20, 0x20, 0xc420047b70, 0x1, 0x1, 0x0, 0xc420162878, 0x158308a)
	/private/tmp/sops-20171011-79678-1wrcwgy/sops-3.0.0/src/go.mozilla.org/sops/sops.go:399 +0x5bd
go.mozilla.org/sops.(*Tree).GenerateDataKeyWithKeyServices(0xc4201c0500, 0xc420047b70, 0x1, 0x1, 0xc4201c4400, 0x5, 0x8, 0x0, 0x0, 0x1681880)
	/private/tmp/sops-20171011-79678-1wrcwgy/sops-3.0.0/src/go.mozilla.org/sops/sops.go:327 +0x214
main.editExample(0x1ab9880, 0xc4201b0a20, 0x1abd280, 0x1b25018, 0x1abd280, 0x1b25018, 0x7fff5fbff741, 0x8, 0x0, 0xc420047b70, ...)
	/private/tmp/sops-20171011-79678-1wrcwgy/sops-3.0.0/src/go.mozilla.org/sops/cmd/sops/edit.go:104 +0x377
main.main.func5(0xc4200b69a0, 0x0, 0x0)
	/private/tmp/sops-20171011-79678-1wrcwgy/sops-3.0.0/src/go.mozilla.org/sops/cmd/sops/main.go:489 +0x1b71
go.mozilla.org/sops/vendor/gopkg.in/urfave/cli%2ev1.HandleAction(0x1639ec0, 0x1752d20, 0xc4200b69a0, 0x0, 0x0)
	/private/tmp/sops-20171011-79678-1wrcwgy/sops-3.0.0/src/go.mozilla.org/sops/vendor/gopkg.in/urfave/cli.v1/app.go:490 +0xd2
go.mozilla.org/sops/vendor/gopkg.in/urfave/cli%2ev1.(*App).Run(0xc420183860, 0xc4200100c0, 0x6, 0x6, 0x0, 0x0)
	/private/tmp/sops-20171011-79678-1wrcwgy/sops-3.0.0/src/go.mozilla.org/sops/vendor/gopkg.in/urfave/cli.v1/app.go:264 +0x635
main.main()
	/private/tmp/sops-20171011-79678-1wrcwgy/sops-3.0.0/src/go.mozilla.org/sops/cmd/sops/main.go:519 +0x201b
```

If I run the command without the `--encryption-context` args, the file is generated.

Using Go 1.8.3 on macOS 10.12

--- a/keyservice/keyservice.go
+++ b/keyservice/keyservice.go
@@ -33,7 +33,7 @@ func KeyFromMasterKey(mk keys.MasterKey) Key {
 			},
 		}
 	case *kms.MasterKey:
-		var ctx map[string]string
+		ctx := make(map[string]string)
 		for k, v := range mk.EncryptionContext {
 			ctx[k] = *v
 		}

--- a/keyservice/server.go
+++ b/keyservice/server.go
@@ -23,7 +23,7 @@ func (ks *Server) encryptWithPgp(key *PgpKey, plaintext []byte) ([]byte, error) 
 }
 
 func (ks *Server) encryptWithKms(key *KmsKey, plaintext []byte) ([]byte, error) {
-	var ctx map[string]*string
+	ctx := make(map[string]*string)
 	for k, v := range key.Context {
 		ctx[k] = &v
 	}
@@ -58,7 +58,7 @@ func (ks *Server) decryptWithPgp(key *PgpKey, ciphertext []byte) ([]byte, error)
 }
 
 func (ks *Server) decryptWithKms(key *KmsKey, ciphertext []byte) ([]byte, error) {
-	var ctx map[string]*string
+	ctx := make(map[string]*string)
 	for k, v := range key.Context {
 		ctx[k] = &v
 	}

